sdk/core/core-client-rest/test/sendRequest.spec.ts

```diff
@@ -27,15 +27,87 @@ describe("sendRequest", () => {
       await sendRequest("POST", mockBaseUrl, mockPipeline, { body: foo });
     });
 
-    it("should handle request body as string", async () => {
+    it("should handle request body as non-json string without content type", async () => {
       const mockPipeline: Pipeline = createEmptyPipeline();
-      const expectedBody = "foo";
+      const expectedBody = '"foo"';
       mockPipeline.sendRequest = async (_client, request) => {
         assert.equal(request.body, expectedBody);
         return { headers: createHttpHeaders() } as PipelineResponse;
       };
       await sendRequest("POST", mockBaseUrl, mockPipeline, { body: "foo" });
     });
+
+    it("should handle request body as json string without content type", async () => {
+      const mockPipeline: Pipeline = createEmptyPipeline();
+      const expectedBody = '{"key":"value"}';
+      mockPipeline.sendRequest = async (_client, request) => {
+        assert.equal(request.body, expectedBody);
+        return { headers: createHttpHeaders() } as PipelineResponse;
+      };
+      await sendRequest("POST", mockBaseUrl, mockPipeline, { body: '{"key":"value"}' });
+    });
+
+    it("should handle request body as non-json string with content type", async () => {
+      const mockPipeline: Pipeline = createEmptyPipeline();
+      const expectedBody = '"foo"';
+      mockPipeline.sendRequest = async (_client, request) => {
+        assert.equal(request.body, expectedBody);
+        return { headers: createHttpHeaders() } as PipelineResponse;
+      };
+      await sendRequest("POST", mockBaseUrl, mockPipeline, {
+        body: "foo",
+        contentType: "application/json",
+      });
+    });
+
+    it("should handle request body as json string with content type", async () => {
+      const mockPipeline: Pipeline = createEmptyPipeline();
+      const expectedBody = JSON.stringify('{"key":"value"}');
+      mockPipeline.sendRequest = async (_client, request) => {
+        assert.equal(request.body, expectedBody);
+        return { headers: createHttpHeaders() } as PipelineResponse;
+      };
+      await sendRequest("POST", mockBaseUrl, mockPipeline, {
+        body: '{"key":"value"}',
+        contentType: "application/json",
+      });
+    });
+
+    it("should handle request body as non-json string with content type in header", async () => {
```

Comment:
I am curious, do other languages allow requests with incompatible body and content-type to be sent as well?

Reply:
this is the current logic, if we see a content type application/json, we will stringify it no matter if it's a valid json string or not.  https://github.com/Azure/azure-sdk-for-js/blob/075c4d1de558fd715a3c450517d6ad28cb360403/sdk/core/core-client-rest/src/sendRequest.ts#L172  

To change this may bring an unnessary breaking to customers. 

From cadl-ranch test coverage dashboard https://azure.github.io/cadl-ranch/ of encde bytes requestBody for test cases like base64 and base64Url and default, at least Python and Java are supporting non-json string with application/json content type. because cadl ranch server backend is expecting application/json in those cases.